File: 10/ruby/Goose.rb
#Goose.rb | Defines the Goose class which inherits attributes and methods
#   from the Bird superclass.
#   #
#   # Begun by: Dr. Adams, for CS 214 at Calvin College.
#   # Completed by:
#   # Date:
#

require './Bird.rb'
require './FlyingBird.rb'

class Goose < FlyingBird

  def call
    "Honk!"
  end

end

File: 10/java/Goose.java
/* Goose.java provides a Goose subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by:
 * Date:
 ******************************************************/

public class Goose extends FlyingBird {

	/* explicit constructor
	 * Receive: name, a String 
	 * PostCond: myName == name.
	 */

	public Goose(String name)
	{
		super(name);
	}



	/* A Goose's Call
	 * Return: a goose-call ("Honk!").
	 */

	public String call()
	{
		return "Honkkkkk!";
	}

} 

